### DISCUSSION TITLE
关于Xray当中路由和探测的问题讨论

### DISCUSSION THREAD

**Discussion by Extreme-Icer** [XTLS/Xray-core#1605] [General]
服务端>>>>>>>>>>>
服务端在禁止回国流量后，是禁止GEOIP:CN出站的，那么，服务端如果将路由策略改为asis，出站的时候会如何？域名嗅探方面，我如果想让客户端传过来的IP还是以IP出站，域名保持不变，我目前将sniffing enable了，routeonly&destoveride也开了，此时会发生什么？
客户端>>>>>>>>>>>
已知Windows端都是HTTP代理，想用socks代理只能另辟蹊径，观察所见，将路由策略设置为asis，开启流量探测，日志几乎都是http的域名，很少有纯IP（一般为tg）。同时使用adguard做广告过滤那么，客户端中outbound freedom 选项选择asis会发生什么？使用adguard是否会使用他的结果？
Android一侧，同样使用adguard，但是它作为vpn接管流量，ng作为socks代理被adguard链接，现有一个问题：ng作为vpn，谷歌各项请求没有问题，adguard作为vpn，谷歌就没有网络，但是网页均可以正常打开。那么开启routeonly自定义配置的ng是不是也会采用adguard的解析结果并发往远端？域名嗅探是哪个DNS完成的呢？NG进阶设置的远程/本地dns又有何作用呢？
Windows端：ADGuard（接管系统DNS解析，开了DOH）——V2rayN（策略：绕过大陆，路由策略asis，流量探测开启、routeonly开启，outbound设置asis或者不设置）
Android端：ADGuard（系统级VPN，开启DOH）——v2rayNG的socks5代理被adguard链接——访问外网，同样绕过大陆，路由asis，routeonly开启。
PS:自定义配置若vpn模式下也能使用，因为有个VPN DNS，谷歌正常解析。Android传给服务端基本都是IP。

@RPRX 还请大佬解释一下其原理，以及服务端/客户端嗅探是用的哪个dns

**Comment by chika0801** [comment]
https://xtls.github.io/document/level-1/routing-lv1-part1.html


>服务端在禁止回国流量后，是禁止GEOIP:CN出站的，那么，服务端如果将路由策略改为asis，出站的时候会如何？

你要发你VPS端配置，看到配置才能进一步分析。ASIS的意思你看下文档就明白了。

>域名嗅探方面，我如果想让客户端传过来的IP还是以IP出站，域名保持不变，我目前将sniffing enable了，routeonly&destoveride也开了，此时会发生什么？

设你客户端发过来的是IP，你在服务端开了sniffing，IP还原成域名，进路由模块，假设你是ASIS，匹配域名规则，没中，这IP匹配IP规则，没中。进出站模块，应该是以入站时的IP出站，VPS去连这IP（我没实验过，你开DEBUG看日志，能自己实验。）

**Comment by chika0801** [comment]
> 已知Windows端都是HTTP代理，想用socks代理只能另辟蹊径，观察所见，将路由策略设置为asis，开启流量探测，日志几乎都是http的域名，很少有纯IP（一般为tg）。同时使用adguard做广告过滤那么，

win下我是用chrome，在用 SwitchyOmega 插入导入http到v2rayn的http端口chrome上用的adg插件过滤广告。adguard插件正常工作，它的流量不受SwitchyOmega影响进v2rayn的。

> 客户端中outbound freedom 选项选择asis会发生什么？

你看v2rayn生成的config你自己会读的话就懂了。asis就直接出，不用你配置里的DNS模块解析这域名，文档上写的。

**Comment by chika0801** [comment]
剩下的 我没用过这种环境，不想帮你分析了，也不熟悉

**Comment by Extreme-Icer** [comment]
服务端配置如下：
```
{
    "log": {
        "loglevel": "debug"
     },
    "routing": {
        "domainStrategy": "AsIs",
	"domainMatcher": "hybrid",
        "rules": [
            {
                "type": "field",
                "ip": [
                    "geoip:cn",
		    "geoip:private"
                ],
                "outboundTag": "block"
            }
        ]
    },
...
            "sniffing": {
                "enabled": false,
                "destOverride": [
                    "http",
                    "tls"
                ],
                "routeOnly": true
            }
        }
    ],
    "outbounds": [
        {
            "protocol": "freedom",
            "tag": "direct"
        },
        {
            "protocol": "blackhole",
            "tag": "block"
        }

```
outbound freedom那个，我没在N和服务端看到明显变化，有时间再看看

**Reply by chika0801** [reply]
你用ctrl+e 这样有格式

你说清楚这部分是客户还是服务端的？帖完整嘛，问题提清楚。outbound freedom那个你是不是在v2rayN的DNS选项界面的选项嘛

**Reply by Extreme-Icer** [reply]
上面说了呀，服务端配置，inbounds部分我去掉了，没有影响
outbound freedom是n设置里面的core:DNS设置。
我N设置的是路由规则asis，然后流量探测+routeonly
NG用的自定义配置，从电脑导出的。日志我还真没看出来outbound freedom那个有啥差别

**Reply by chika0801** [reply]
这不是tg群的及时帮助，你尽量回的时候帖上配置，问题说清楚，有人看到一次性看完需要的消息，快点帮你，不然非要打字对话解决效率倒是很低

**Reply by chika0801** [reply]
> 上面说了呀，服务端配置，inbounds部分我去掉了，没有影响 outbound freedom是n设置里面的core:DNS设置。 我N设置的是路由规则asis，然后流量探测+routeonly NG用的自定义配置，从电脑导出的。日志我还真没看出来outbound freedom那个有啥差别

看你配置，你关了 sniffing 你看是false的都没嗅探。

**Reply by Extreme-Icer** [reply]
服务端我没开嗅探，直接正常传。不过就是路由asis了。客户端开嗅探，Android因为是socks代理被adguard连接，日志体现的全是socks传入。Windows全是域名然后传给了服务端

**Reply by chika0801** [reply]
> 服务端我没开嗅探，直接正常传。不过就是路由asis了。客户端开嗅探，Android因为是socks代理被adguard连接，日志体现的全是socks传入。Windows全是域名然后传给了服务端

Android因为是socks代理被adguard连接，日志体现的全是socks传入。Windows全是域名然后传给了服务端

这部分我没这得用过，帮不上

**Reply by Extreme-Icer** [reply]
行 谢谢大佬

**Comment by chika0801** [comment]
根据你帖的配置，设sniffing 你开，routeOnly也是true你是用的ASIS，你客户端发过来是IP

**https://github.com/XTLS/Xray-core/discussions/1605#discussioncomment-4883258**

这里就回答了，再编辑下

设你客户端发过来的是IP，你在服务端开了sniffing，IP还原成域名，进路由模块，你是ASIS，匹配域名规则，没中，这IP匹配IP规则，没中。因为你是routeOnly true，进出站模块的请求是IP，应该是以入站时的IP出站，VPS去连这IP（我没实验过，你开DEBUG看日志，能自己实验。）

**Comment by chika0801** [comment]
你研究原理 客户端和服务端开Debug看日志，你要搞懂Xray这边流程也有那个推荐文章，剩下就自己试了

**Comment by chika0801** [comment]
你不是在群里问过嘛，想起来了

**Reply by Extreme-Icer** [reply]
不过我是想请rprx大佬解读一下这个路由和探测模块23333

**Reply by chika0801** [reply]
> 不过我是想请rprx大佬解读一下这个路由和探测模块23333

明显大佬太忙，你发讨论区更是一般没人看

**Reply by Extreme-Icer** [reply]
确实

**Comment by chika0801** [comment]
那句 简单是最好

Windows端：ADGuard（接管系统DNS解析，开了DOH）——V2rayN（策略：绕过大陆，路由策略asis，流量探测开启、routeonly开启，outbound设置asis或者不设置）
Android端：ADGuard（系统级VPN，开启DOH）——v2rayNG的socks5代理被adguard链接——访问外网，同样绕过大陆，路由asis，routeonly开启。

建议你没搞这么复杂了。。。

**Reply by Extreme-Icer** [reply]
广告过滤需求，没法
